Examples/create-app.js

```diff
@@ -260,47 +256,78 @@ function androidSetup() {
+        fs.writeFileSync(buildGradlePath, buildGradleContents);
+    }
 
     let settingsGradleContents = fs.readFileSync(settingsGradlePath, "utf8");
     const settingsGradleInclude = "include \':app\'";
-    const codePushProjectImport= `':react-native-code-push'\nproject(':react-native-code-push').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-code-push/android/app')`;
+    const codePushProjectImport = `':react-native-code-push'\nproject(':react-native-code-push').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-code-push/android/app')`;
     settingsGradleContents = settingsGradleContents.replace(settingsGradleInclude,
         `${settingsGradleInclude}, ${codePushProjectImport}`);
     fs.writeFileSync(settingsGradlePath, settingsGradleContents);
 
-    const getJSBundleFileOverride = `
-    @Override
-    protected String getJSBundleFile(){
-        return CodePush.getJSBundleFile();
-    }
-    `;
+    let importCodePush = `\nimport com.microsoft.codepush.react.CodePush;`;
+    let reactNativeHostInstantiationImport = "import android.app.Application;";
     let mainApplicationContents = fs.readFileSync(mainApplicationPath, "utf8");
-    const reactNativeHostInstantiation = "new ReactNativeHost(this) {";
+    let getJSBundleFileOverride = "";
+    let reactNativeHostInstantiation = "";
+
+    // handle react-native version with java
+    if (reactNativeVersionIsLowerThanV073) {
+        reactNativeHostInstantiation = "new ReactNativeHost(this) {";
+
+        getJSBundleFileOverride = `
+        @Override
+        protected String getJSBundleFile(){
+            return CodePush.getJSBundleFile();
+        }
+        `;
+    }
+    // handle react-native version with kotlin
+    else {
+        reactNativeHostInstantiation = "object : DefaultReactNativeHost(this) {"
+        getJSBundleFileOverride = `
+         override fun getJSBundleFile(): String {
+            return CodePush.getJSBundleFile() 
+        }
+        `;
+        importCodePush = importCodePush.replace(';', '');
+        reactNativeHostInstantiationImport = reactNativeHostInstantiationImport.replace(';', '');
+    }
     mainApplicationContents = mainApplicationContents.replace(reactNativeHostInstantiation,
         `${reactNativeHostInstantiation}${getJSBundleFileOverride}`);
 
-    const importCodePush = `\nimport com.microsoft.codepush.react.CodePush;`;
-    const reactNativeHostInstantiationImport = "import android.app.Application;";
     mainApplicationContents = mainApplicationContents.replace(reactNativeHostInstantiationImport,
         `${reactNativeHostInstantiationImport}${importCodePush}`);
     fs.writeFileSync(mainApplicationPath, mainApplicationContents);
 }
 
+function getAppDelegateName() { return fs.readdirSync(path.join('ios', appName)).find(file => file.endsWith('mm') || file.endsWith('m')) };
```

Comment:
can we make it more explicit? It may break if RN start adding more `mm/m` files.